Library/Homebrew/build.rb

```diff
@@ -83,7 +83,8 @@ def install
       ENV.keg_only_deps = keg_only_deps
       ENV.deps = formula_deps
       ENV.run_time_deps = run_time_deps
-      ENV.x11 = reqs.any? { |rq| rq.is_a?(X11Requirement) }
+      ENV.x11 = reqs.any? { |rq| rq.is_a?(X11Requirement) } ||
```

Comment:
I've been wondering if we should do that for a while now. For the moment, I think it's better not to, and because of how the requirement seemingly gets pruned out it won't have an effect in this context.

I do think we need to revisit the XorgRequirement (it was my first non-trivial Ruby code, which might explain some of the poor implementation and design details) but IMO we should do it:

a) In a separate PR where we can test it more fully
b) In light of the fact that Homebrew will likely want it to be completely rewritten when we eventually send it upstream.

Furthermore, I don't think we should do that upstream work any time soon. Even though I started with them, requirements are starting to look like a bad place to do upstream work since they're necessarily tied to the official LInuxbrew taps - ties that upstream isn't fully comfortable with.

Mike made a good suggestion for work I could do in the future to make this tying less problematic from their POV. See https://github.com/Homebrew/brew/pull/2274#discussion_r106812068 and its follow-up comments.